src/App.jsx

```diff
@@ -1,17 +1,38 @@
+import { useState } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
+import messages from './data/messages.json';
 
 const App = () => {
+  const [entries, setEntries] = useState(
+    messages.map((entry) => ({...entry}))
+  );
+
+  const toggleLike = (id) => {
+    const updatedEntries = entries.map((entry) =>
+      entry.id === id ? { ...entry, liked: !entry.liked } : entry
+    );
+    setEntries(updatedEntries);
+  };
```

Comment:
Nice work on the ternary here! This is succinct and still readable since you are just performing a basic check!